File: CoreTransformer/FlowPathway.vb
'************************************************************************************
' STSimStockFlow: A .NET library for simulating stocks and flows
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

Class FlowPathway

    Private m_Iteration As Nullable(Of Integer)
    Private m_Timestep As Nullable(Of Integer)
    Private m_FromStratumId As Nullable(Of Integer)
    Private m_FromStateClassId As Nullable(Of Integer)
    Private m_FromMinimumAge As Nullable(Of Integer)
    Private m_FromStockTypeId As Integer
    Private m_ToStratumId As Nullable(Of Integer)
    Private m_ToStateClassId As Nullable(Of Integer)
    Private m_ToMinimumAge As Nullable(Of Integer)
    Private m_ToStockTypeId As Integer
    Private m_TransitionGroupId As Integer
    Private m_StateAttributeTypeId As Nullable(Of Integer)
    Private m_FlowTypeId As Integer
    Private m_Multiplier As Double
    Private m_FlowAmount As Double

    Public Sub New(
        ByVal iteration As Nullable(Of Integer),
        ByVal timestep As Nullable(Of Integer),
        ByVal fromStratumId As Nullable(Of Integer),
        ByVal fromStateClassId As Nullable(Of Integer),
        ByVal fromMinimumAge As Nullable(Of Integer),
        ByVal fromStockTypeId As Integer,
        ByVal toStratumId As Nullable(Of Integer),
        ByVal toStateClassId As Nullable(Of Integer),
        ByVal toMinimumAge As Nullable(Of Integer),
        ByVal toStockTypeId As Integer,
        ByVal transitionGroupId As Integer,
        ByVal stateAttributeTypeId As Nullable(Of Integer),
        ByVal flowTypeId As Integer,
        ByVal multiplier As Double)

        Me.m_Iteration = iteration
        Me.m_Timestep = timestep
        Me.m_FromStratumId = fromStratumId
        Me.m_FromStateClassId = fromStateClassId
        Me.m_FromMinimumAge = fromMinimumAge
        Me.m_FromStockTypeId = fromStockTypeId
        Me.m_ToStratumId = toStratumId
        Me.m_ToStateClassId = toStateClassId
        Me.m_ToMinimumAge = toMinimumAge
        Me.m_ToStockTypeId = toStockTypeId
        Me.m_TransitionGroupId = transitionGroupId
        Me.m_StateAttributeTypeId = stateAttributeTypeId
        Me.m_FlowTypeId = flowTypeId
        Me.m_Multiplier = multiplier

    End Sub

    Public ReadOnly Property Iteration As Nullable(Of Integer)
        Get
            Return Me.m_Iteration
        End Get
    End Property

    Public ReadOnly Property Timestep As Nullable(Of Integer)
        Get
            Return Me.m_Timestep
        End Get
    End Property

    Public ReadOnly Property FromStratumId As Nullable(Of Integer)
        Get
            Return Me.m_FromStratumId
        End Get
    End Property

    Public ReadOnly Property FromStateClassId As Nullable(Of Integer)
        Get
            Return Me.m_FromStateClassId
        End Get
    End Property

    Public ReadOnly Property FromMinimumAge As Nullable(Of Integer)
        Get
            Return Me.m_FromMinimumAge
        End Get
    End Property

    Public ReadOnly Property FromStockTypeId As Integer
        Get
            Return Me.m_FromStockTypeId
        End Get
    End Property

    Public ReadOnly Property ToStratumId As Nullable(Of Integer)
        Get
            Return Me.m_ToStratumId
        End Get
    End Property

    Public ReadOnly Property ToStateClassId As Nullable(Of Integer)
        Get
            Return Me.m_ToStateClassId
        End Get
    End Property

    Public ReadOnly Property ToMinimumAge As Nullable(Of Integer)
        Get
            Return Me.m_ToMinimumAge
        End Get
    End Property

    Public ReadOnly Property ToStockTypeId As Integer
        Get
            Return Me.m_ToStockTypeId
        End Get
    End Property

    Public ReadOnly Property TransitionGroupId As Integer
        Get
            Return Me.m_TransitionGroupId
        End Get
    End Property

    Public ReadOnly Property StateAttributeTypeId As Nullable(Of Integer)
        Get
            Return Me.m_StateAttributeTypeId
        End Get
    End Property

    Public ReadOnly Property FlowTypeId As Integer
        Get
            Return Me.m_FlowTypeId
        End Get
    End Property

    Public ReadOnly Property Multiplier As Double
        Get
            Return Me.m_Multiplier
        End Get
    End Property

    Public Property FlowAmount As Double
        Get
            Return Me.m_FlowAmount
        End Get
        Set(value As Double)
            Me.m_FlowAmount = value
        End Set
    End Property

End Class

File: Runtime/FlowPathway.cs
//*********************************************************************************************
// STSimStockFlow: A SyncroSim Module for the ST-Sim Stocks and Flows Add-In.
//
// Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
//
//*********************************************************************************************

namespace SyncroSim.STSimStockFlow
{
	internal class FlowPathway
	{
		private int? m_Iteration;
		private int? m_Timestep;
		private int? m_FromStratumId;
		private int? m_FromStateClassId;
		private int? m_FromMinimumAge;
		private int m_FromStockTypeId;
		private int? m_ToStratumId;
		private int? m_ToStateClassId;
		private int? m_ToMinimumAge;
		private int m_ToStockTypeId;
		private int m_TransitionGroupId;
		private int? m_StateAttributeTypeId;
		private int m_FlowTypeId;
		private double m_Multiplier;
		private double m_FlowAmount;

		public FlowPathway(int? iteration, int? timestep, int? fromStratumId, int? fromStateClassId, int? fromMinimumAge, int fromStockTypeId, int? toStratumId, int? toStateClassId, int? toMinimumAge, int toStockTypeId, int transitionGroupId, int? stateAttributeTypeId, int flowTypeId, double multiplier)
		{
			this.m_Iteration = iteration;
			this.m_Timestep = timestep;
			this.m_FromStratumId = fromStratumId;
			this.m_FromStateClassId = fromStateClassId;
			this.m_FromMinimumAge = fromMinimumAge;
			this.m_FromStockTypeId = fromStockTypeId;
			this.m_ToStratumId = toStratumId;
			this.m_ToStateClassId = toStateClassId;
			this.m_ToMinimumAge = toMinimumAge;
			this.m_ToStockTypeId = toStockTypeId;
			this.m_TransitionGroupId = transitionGroupId;
			this.m_StateAttributeTypeId = stateAttributeTypeId;
			this.m_FlowTypeId = flowTypeId;
			this.m_Multiplier = multiplier;
		}

		public int? Iteration
		{
			get
			{
				return this.m_Iteration;
			}
		}

		public int? Timestep
		{
			get
			{
				return this.m_Timestep;
			}
		}

		public int? FromStratumId
		{
			get
			{
				return this.m_FromStratumId;
			}
		}

		public int? FromStateClassId
		{
			get
			{
				return this.m_FromStateClassId;
			}
		}

		public int? FromMinimumAge
		{
			get
			{
				return this.m_FromMinimumAge;
			}
		}

		public int FromStockTypeId
		{
			get
			{
				return this.m_FromStockTypeId;
			}
		}

		public int? ToStratumId
		{
			get
			{
				return this.m_ToStratumId;
			}
		}

		public int? ToStateClassId
		{
			get
			{
				return this.m_ToStateClassId;
			}
		}

		public int? ToMinimumAge
		{
			get
			{
				return this.m_ToMinimumAge;
			}
		}

		public int ToStockTypeId
		{
			get
			{
				return this.m_ToStockTypeId;
			}
		}

		public int TransitionGroupId
		{
			get
			{
				return this.m_TransitionGroupId;
			}
		}

		public int? StateAttributeTypeId
		{
			get
			{
				return this.m_StateAttributeTypeId;
			}
		}

		public int FlowTypeId
		{
			get
			{
				return this.m_FlowTypeId;
			}
		}

		public double Multiplier
		{
			get
			{
				return this.m_Multiplier;
			}
		}

		public double FlowAmount
		{
			get
			{
				return this.m_FlowAmount;
			}
			set
			{
				this.m_FlowAmount = value;
			}
		}
	}
}